facturacion_mexico/facturacion_fiscal/timbrado_api.py

```
@@ -3193,10 +3218,116 @@ def create_substitution_si(si_name: str):
+#   - FAST_WINDOW: durante los primeros minutos tras timbrar B (periodo crítico) se permite un
+#     intento por tick de 1 min (recuperación rápida del 404 transitorio prolongado).
+#   - SLOW_INTERVAL: pasada la ventana rápida, solo se reintenta si transcurrieron ≥5 min desde el
+#     último contacto con el PAC (fm_last_pac_sync). Esto escalona a ~1 intento/5 min y evita
+#     ~120 DELETE para un caso prolongado, sin campos ni contadores nuevos.
+_SUBSTITUTION_CANCEL_MAX_AGE_MIN = 120  # corte total (~2 h desde fecha_timbrado de B)
+_SUBSTITUTION_CANCEL_FAST_WINDOW_MIN = 5  # ventana rápida inicial: 1 intento por minuto
+_SUBSTITUTION_CANCEL_SLOW_INTERVAL_MIN = 5  # después: 1 intento cada ~5 min
+_SUBSTITUTION_CANCEL_BATCH = 50
+
+
+def _complete_documental_cancellation(orig_ffm_name: str, orig_si_name: str | None = None) -> dict:
```

Ruff (0.15.21)

[warning] Too many branches (13 > 12)

(PLR0912)